src/Domain/LeanCode.DomainModels.EF/TimestampTzLocalTimestampExpressionInterceptor.cs

```diff
@@ -0,0 +1,56 @@
+using System.Linq.Expressions;
+using System.Reflection;
+using LeanCode.DomainModels.Model;
+using Microsoft.EntityFrameworkCore.Diagnostics;
+
+namespace LeanCode.DomainModels.EF;
+
+public sealed class TimestampTzLocalTimestampExpressionInterceptor : IQueryExpressionInterceptor
+{
+    private sealed class TimestampTzLocalTimestampExpressionVisitor : ExpressionVisitor
+    {
+        private static readonly Type TimestampTzType = typeof(TimestampTz);
+        private static readonly Type TimeZoneInfoType = typeof(TimeZoneInfo);
+
+        private static readonly PropertyInfo LocalTimestampProperty =
+            TimestampTzType.GetProperty(nameof(TimestampTz.LocalTimestamp))
+            ?? throw new MissingMemberException(TimestampTzType.FullName, nameof(TimestampTz.LocalTimestamp));
+
+        private static readonly PropertyInfo UtcTimestampProperty =
+            TimestampTzType.GetProperty(nameof(TimestampTz.UtcTimestamp))
+            ?? throw new MissingMemberException(TimestampTzType.FullName, nameof(TimestampTz.UtcTimestamp));
+
+        private static readonly PropertyInfo TimeZoneIdProperty =
+            TimestampTzType.GetProperty(nameof(TimestampTz.TimeZoneId))
+            ?? throw new MissingMemberException(TimestampTzType.FullName, nameof(TimestampTz.TimeZoneId));
+
+        private static readonly MethodInfo ConvertTimeBySystemTimeZoneIdMethod =
+            TimeZoneInfoType.GetMethod(
+                nameof(TimeZoneInfo.ConvertTimeBySystemTimeZoneId),
+                new[] { typeof(DateTime), typeof(string) }
+            )
+            ?? throw new MissingMemberException(
+                TimeZoneInfoType.FullName,
+                nameof(TimeZoneInfo.ConvertTimeBySystemTimeZoneId)
+            );
+
+        protected override Expression VisitMember(MemberExpression node)
+        {
+            return node.Member == LocalTimestampProperty
+                ? Expression.Call(
```

Comment:
This works for both SQL and postgres?

Reply:
Not sure about SQL Server, it's not the target of this PR but if we wanted to explicitly support it, we could provide a translation for `GetLocalTimestampWithOffset` using [EF.Functions.AtTimeZone](https://learn.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.sqlserverdbfunctionsextensions.attimezone?view=efcore-7.0#microsoft-entityframeworkcore-sqlserverdbfunctionsextensions-attimezone(microsoft-entityframeworkcore-dbfunctions-system-datetime-system-string)).

Reply:
If we put it in non-Postgres specific package, we should support SQL Server (someone will use it there, and be very surprised that it does not work).

Reply:
That should be doable, but… can we somehow recognize which provider is used from inside our interceptor? If not, we would probably have to split them into two versions, one for SQL Server and one for Postgres :weary:

Reply:
I'm pretty sure it is possible, but probably not officially supported ^^'

I think two separate extension methods would be OK.

Reply:
@marcin-dardzinski We have decided that there will be no support for SQL Server at this time (and the type has a comment explaining that it's meant to be used only with PostgreSQL).